### PR TITLE
Localization of encryption keys

### PR DESCRIPTION
**Pull request by chavonadelal**
## About the PR

Localization of encryption keys
## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: to4no_fix

### CODE REVIEW

**Review by TheShuEd — CHANGES_REQUESTED** [review]
Procedural generation of localization keys like here is a very awkward thing to support. It is better to put LocId field in the communication channel prototype and use it directly

**Comment by chavonadelal** [comment]
> Procedural generation of localization keys like here is a very awkward thing to support. It is better to put LocId field in the communication channel prototype and use it directly

@TheShuEd If I understood you correctly and you suggest putting the localization key directly into the ID of the communication channel, then this does not seem possible. Errors appear.
image

**Comment by TheShuEd** [comment]
Yep, you need change DataField type, and adapt code for it

**Comment by TheShuEd** [comment]
Wait, no, dont toich Id filed

**Comment by TheShuEd** [comment]
Just... Just use name field?

**Comment by chavonadelal** [comment]
> Just... Just use name field?

@TheShuEd I don't think this will work since the default channel is taken from the prototype of the encryption key where the default channel ID is specified. As I understand it.
image

File: Content.Shared/Radio/EntitySystems/EncryptionKeySystem.cs [hunk 1]
```diff
@@ -239,14 +239,14 @@ public void AddChannelsExamine(HashSet<string> channels, string? defaultChannel,
             {
                 var msg = Loc.GetString("examine-headset-default-channel",
                 ("prefix", SharedChatSystem.DefaultChannelPrefix),
-                ("channel", defaultChannel),
+                ("channel", Loc.GetString($"chat-radio-{defaultChannel.ToString().ToLower()}")),
```
Comment (Tayrtahn) [inline Content.Shared/Radio/EntitySystems/EncryptionKeySystem.cs:242]:
```suggestion
                ("channel", proto.LocalizedName),
```

File: Content.Shared/Radio/EntitySystems/EncryptionKeySystem.cs [hunk 2]
```diff
@@ -239,14 +239,14 @@ public void AddChannelsExamine(HashSet<string> channels, string? defaultChannel,
...
                 ("color", proto.Color));
                 examineEvent.PushMarkup(msg);
             }
             if (HasComp<EncryptionKeyComponent>(examineEvent.Examined))
             {
                 var msg = Loc.GetString("examine-encryption-default-channel",
-                ("channel", defaultChannel),
+                ("channel", Loc.GetString($"chat-radio-{defaultChannel.ToString().ToLower()}")),
```
Comment (Tayrtahn) [inline Content.Shared/Radio/EntitySystems/EncryptionKeySystem.cs:249]:
```suggestion
                ("channel", proto.LocalizedName),
```

**Review by Tayrtahn — CHANGES_REQUESTED** [review]
Building localized string IDs from prototype IDs like this is pretty janky. RadioChannelPrototype actually has a LocalizedName field; just use that.

My suggestions are untested, but I think they should work.

Also do what Ed said and change RadioChannelPrototype.Name from string to LocId.

**Comment by chavonadelal** [comment]
> Building localized string IDs from prototype IDs like this is pretty janky. RadioChannelPrototype actually has a LocalizedName field; just use that.
> 
> My suggestions are untested, but I think they should work.
> 
> Also do what Ed said and change RadioChannelPrototype.Name from string to LocId.

I did what you asked 😄

**Review by TheShuEd — APPROVED** [review]

**Review by Tayrtahn — APPROVED** [review]
Looks good to me. I'm going to assume you've tested it and it works.

**Comment by chavonadelal** [comment]
> Looks good to me. I'm going to assume you've tested it and it works.

Yes, the required text appears, I didn’t notice any errors.

**Pull request merged**